publication-model/src/main/java/no/unit/nva/model/associatedartifacts/file/UploadedFile.java

```diff
@@ -0,0 +1,45 @@
+package no.unit.nva.model.associatedartifacts.file;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import java.util.UUID;
+
+@JsonTypeInfo(use = Id.NAME, property = "type")
+@JsonTypeName(UploadedFile.TYPE)
+public class UploadedFile extends File {
+
+    public static final String TYPE = "UploadedFile";
+
+    @JsonCreator
+    public UploadedFile(@JsonProperty(IDENTIFIER_FIELD) UUID identifier,
+                        @JsonProperty(NAME_FIELD) String name,
+                        @JsonProperty(MIME_TYPE_FIELD) String mimeType,
```

Comment:
Do we have restrictions on valid mime types?

Reply:
I suspect a mime-type is a complex object. We should discuss this.

Name can be a value object too.

Reply:
It is file from datamodel which has contains information for uploaded file. We should probably validate what user sends in request body in CreateUploadHandler, so we do not allow invalid mimetypes/too large files?